server/helpers.py

```diff
@@ -0,0 +1,876 @@
+    videoModel = self.model('video', 'video')
+
+    file = getFile(self, fileId, user=user)
+
+    vData = getVideoData(
+        self, fileId=fileId, formatId=formatId, user=user, ensure=True)
+
+    vJob = videoModel.findOne({
+        'type': VideoEnum.JOB,
+        'jobType': VideoEnum.ANALYSIS,
+        'fileId': vData['fileId'],
+        'sourceFileId': vData['sourceFileId']
+    })
+
+    jobId = (vJob or {}).get('jobId')
+
+    itemAlreadyProcessed = False
+    job = None
+    if jobId is not None:
+        job = jobModel.load(jobId, level=AccessType.READ, user=user)
+
+    if not force:
+        if job is not None:
+            status = job['status']
+            if status not in (
+                    None, JobStatus.ERROR, JobStatus.CANCELED):
+                itemAlreadyProcessed = True
+
+        if itemAlreadyProcessed:
+            result = {
+                'video': {
+                    'jobCreated': False,
+                    'message': 'Analysis job already created.'
+                }
+            }
+
+            result.update(job)
+            return result
+
+    # if we are *re*running an analysis job (force=True), remove all data
+    # for this video that were created by the last analysis job, and all data
+    # create by jobs that depend on the results of the analysis job
+    if force:
+        cancelVideoJobs(
+            self, fileId, formatId=formatId, user=user)
+        removeVideoData(
+            self, fileId, formatId=formatId, user=user)
+
+    # begin construction of the actual job
+    if not userToken:
+        userToken = tokenModel.createToken(
+            user, days=1, scope=TokenScope.USER_AUTH)
+
+    stash = itemModel.createHiddenItem(user)
+    stash = itemModel.load(stash['_id'], level=AccessType.WRITE, user=user)
+    targetFile = getFile(self, vData['fileId'], user=user)
+
+    jobTitle = '[video] Analysis'
+    job = jobModel.createJob(
+        title=jobTitle,
+        type='video',
+        user=user,
+        handler='worker_handler'
+    )
+    jobToken = jobModel.createJobToken(job)
+
+    job['kwargs'] = job.get('kwargs', {})
+    job['kwargs']['task'] = {
+        'mode': 'docker',
+
+        # TODO(opadron): replace this once we have a maintained
+        #                image on dockerhub
+        'docker_image': 'ffmpeg_local',
+        'pull_image': False,
+        'container_args': ['analyze'],
+        'inputs': [
+            {
+                'id': 'input',
+                'type': 'string',
+                'format': 'text',
+                'target': 'filepath'
+            }
+        ],
+        'outputs': [
+            {
+                'id': 'meta',
+                'type:': 'string',
+                'format': 'text',
+                'target': 'filepath',
+                'path': '/mnt/girder_worker/data/meta.json'
+            },
+        ]
+    }
+
+    _, itemExt = os.path.splitext(targetFile['name'])
+
+    job['kwargs']['inputs'] = {
```

Comment:
Just as an FYI - this approach to job creation in girder is going to be deprecated and eventually removed. There is a much easier/cleaner way of handling docker tasks

Reply:
Thanks for the heads up.  Might you have a pointer to some documentation I can look through?  It might be a while before I have a chance to get back to this project, but I suspect it might help inform some of my decisions on the resonantgeo work.

Reply:
yeah, i 'm working on comprehensive documentation now.  The best resource in the mean time is the [HACKING](https://github.com/girder/girder_worker/blob/master/tests/integration/HACKING.md) and [TRANSITION](https://github.com/girder/girder_worker/blob/transition-document/TRANSITION.md) documents